fence/agents/azure_arm/fence_azure_arm.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import atexit
+sys.path.append("/usr/share/fence")
+from fencing import *
+from fencing import fail, fail_usage, EC_TIMED_OUT, run_delay
+
+#BEGIN_VERSION_GENERATION
+RELEASE_VERSION="4.0.24.6-7e576"
+BUILD_DATE="(built Thu Nov 3 15:43:03 STD 2016)"
+REDHAT_COPYRIGHT="Copyright (C) Red Hat, Inc. 2004-2010 All rights reserved."
+#END_VERSION_GENERATION
+
+compute_client = None
+
+def get_nodes_list(conn, options):
+    result = {}
+    if compute_client:
+        rgName = options["--resourceGroup"]
+        vms = compute_client.virtual_machines.list(rgName)
+        for vm in vms:
+            result[vm.name] = ("", None)
+
+    return result
+
+def get_power_status(conn, options):
+    logging.info("getting power status for VM " + options["--plug"])
+
+    if compute_client:
+        rgName = options["--resourceGroup"]
+        vmName = options["--plug"]
+
+        powerState = "unknown"
+        vmStatus = compute_client.virtual_machines.get(rgName, vmName, "instanceView")
+        for status in vmStatus.instance_view.statuses:
+            if status.code.startswith("PowerState"):
+                powerState = status.code
+                break
+
+        logging.info("Found power state of VM: " + powerState)
+        if powerState == "PowerState/running":
+            return "on"
+
+    return "off"
+
+def set_power_status(conn, options):
+    logging.info("setting power status for VM " + options["--plug"] + " to " + options["--action"])
+
+    if compute_client:
+        rgName = options["--resourceGroup"]
+        vmName = options["--plug"]
+
+        if (options["--action"]=="off"):
+            logging.info("Deallocating " + vmName + "in resource group " + rgName)
+            compute_client.virtual_machines.deallocate(rgName, vmName)
+        elif (options["--action"]=="reboot"):
+            logging.info("Restarting " + vmName + "in resource group " + rgName)
+            compute_client.virtual_machines.restart(rgName, vmName)
+
+        sys.exit(0)
+
+    sys.exit(1)
+
+def define_new_opts():
+    all_opt["resourceGroup"] = {
+        "getopt" : "rg:",
+        "longopt" : "resourceGroup",
+        "help" : "-rg, --resourceGroup=[name]       Name of the resource group",
+        "shortdesc" : "Name of resource group.",
+        "required" : "1",
+        "order" : 2
+    }
+    all_opt["tenantId"] = {
+        "getopt" : "tid:",
+        "longopt" : "tenantId",
+        "help" : "-tid, --tenantId=[name]       Id of the Azure Active Directory tenant",
+        "shortdesc" : "Id of Azure Active Directory tenant.",
+        "required" : "1",
+        "order" : 3
+    }
+    all_opt["subscriptionId"] = {
+        "getopt" : "sid:",
+        "longopt" : "subscriptionId",
+        "help" : "-sid, --subscriptionId=[name]       Id of the Azure subscription",
+        "shortdesc" : "Id of the Azure subscription.",
+        "required" : "1",
+        "order" : 4
+    }
+
+# Main agent method
+def main():
+    global compute_client
```

Comment:
I would prefer to pass compute_client as an arg (instead of conn in other agents) but I can live with that :)